src/components/QrScanner/QrMobile.tsx

```diff
@@ -0,0 +1,291 @@
+import React from "react";
+import { QrProps } from "./types";
+import { useEffect, useState } from "react";
+import { CirclePlayIcon } from "lucide-react";
+import { CircleAlert } from "lucide-react";
+import { CheckCircle2 } from "lucide-react";
+import { ChevronsUp } from "lucide-react";
+import { QrReader } from "react-qr-reader";
+import { REGISTRATION_STATUS } from "@/constants/registrations";
+import NoCamera from "../../../public/assets/icons/nocamera_icon.svg";
+import { Result } from "@zxing/library";
+import { AnimatePresence, motion } from "framer-motion";
+import Image from "next/image";
+
+// an enumeration for the stages of QR code scanning
+const QR_SCAN_STAGE = {
+  SCANNING: "SCANNING",
+  FAILED: "FAILED",
+  SUCCESS: "SUCCESS",
+};
+
+// facing mode for the camera
+const CAMERA_FACING_MODE = {
+  FRONT: "user",
+  BACK: "environment",
+};
```

Comment:
With these duplicated constants and a few other duplicate functions in the QrMobile class, at a glance it looks like there's a lot of duplicate code across the Mobile / Desktop components. I'm wondering how much we had to change between the Mobile / Desktop components? Was it just the styling? Were there any logic changes?

If it was mainly styling, it might be worth consolidating these two components together and trying out Tailwind Breakpoints for UI responsiveness to avoid increasing the long-term maintenance effort of the QR component - I know my message in Slack suggested putting this off until future PRs, but wanted to check with you if this could possibly be a small fix on your side?

Reply:
there's a small amount of logic involved, but I'll give it a try. The mobile/desktop versions vary in the design which would lead to a bit of code repetition but I'm assuming it would be a lot less than separate components

Reply:
done!